src/Utilities/FlowAnalysis/FlowAnalysis/Analysis/ParameterValidationAnalysis/ParameterValidationAnalysis.ParameterValidationDataFlowOperationVisitor.cs

```diff
@@ -225,6 +225,13 @@ public override ParameterValidationAbstractValue Visit(IOperation operation, obj
                 return value;
             }
 
+            public override ParameterValidationAbstractValue VisitReDimClause(IReDimClauseOperation operation, object? argument)
+            {
+                var value = base.VisitReDimClause(operation, argument);
+                MarkValidatedLocations(operation.Operand);
+                return value;
+            }
```

Comment:
@Evangelink Can you try if reverting the `VisitReDimClause` override in this file has any effect? I think your changes to the next two files might make this override redundant. If not, then feel free to retain this override.

Reply:
Tests `ReDim_FirstInstruction_NoDiagnostic` and `ReDim_FirstInstructionMultipleVariables_NoDiagnostic` are failing (they report the diagnostic while they shouldn't) if I revert this change.

Reply:
No worries - your change LGTM. Thanks!